lib/ProductOpener/Display.pm

```diff
@@ -4688,18 +4691,32 @@ sub add_params_to_query($request_ref, $query_ref) {
 }
 
 
-=head2 initialize_knowledge_panels_options( $knowledge_panels_options_ref )
+=head2 initialize_knowledge_panels_options( $knowledge_panels_options_ref, $request_ref )
 
 Initialize the options for knowledge panels from parameters.
 
 =cut
 
-sub initialize_knowledge_panels_options($knowledge_panels_options_ref) {
+sub initialize_knowledge_panels_options($knowledge_panels_options_ref, $request_ref) {
 
 	# Activate physical activity knowledge panel only when specified
 	if (single_param("activate_knowledge_panel_physical_activities")) {
 		$knowledge_panels_options_ref->{activate_knowledge_panel_physical_activities} = 1;
 	}
+
+	# Specify if we knowledge panels are requested from the app or the website
+	# in order to allow different behaviours (e.g. showing ingredients before nutrition on the web)
+	# possible values: "web", "app"
+	my $knowledge_panels_client = single_param("knowledge_panels_client");
+	if ((not defined $knowledge_panels_client) or (($knowledge_panels_client ne "web")) and ($knowledge_panels_client ne "app")) {
```

Comment:
reading this conditions make my head hurts :-D

I'm not sure parenthesis are set the way you want !
```suggestion
       # set a default value if client is not defined to app or web
	if ((not defined $knowledge_panels_client) or (($knowledge_panels_client ne "web") and ($knowledge_panels_client ne "app"))) {
```

Reply:
This is better indeed, thanks